src/dynamic_honey_badger/votes.rs

```diff
@@ -55,8 +55,8 @@ where
             voter: voter.clone(),
             sig: self.netinfo.secret_key().sign(ser_vote),
         };
-        self.pending.insert(voter.clone(), signed_vote);
-        Ok(self.pending.get(&voter).expect("entry was just inserted"))
+        self.pending.remove(&voter);
+        Ok(self.pending.entry(voter).or_insert(signed_vote))
```

Comment:
I think this is correct, unless someone can think of a use-case where it would be really helpful to notify the caller that he did try to insert things twice?

Reply:
I guess this is fine then =).